data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/PeerForwardingProcessingDecoratorTest.java

```diff
@@ -115,6 +116,29 @@ void PeerForwardingProcessingDecorator_execute_should_forwardRecords_with_correc
             assertThat(records, equalTo(testData));
         }
 
+        @Test
+        void PeerForwardingProcessingDecorator_execute_should_receiveRecords() {
+            Collection<Record<Event>> forwardTestData = Collections.singletonList(record);
+            Collection<Record<Event>> receiveTestData = Collections.singletonList(mock(Record.class));
+
+            when(peerForwarder.forwardRecords(forwardTestData)).thenReturn(forwardTestData);
+            when(peerForwarder.receiveRecords()).thenReturn(receiveTestData);
+
+            final Collection<Record<Event>> recordsToProcessLocally = CollectionUtils.union(forwardTestData, receiveTestData);
```

Comment:
It might be more readable to name this `recordsExpectedLocal` or somehow include "expected" in the name.